atomic_reactor/plugin.py

```diff
@@ -210,6 +214,9 @@ def __init__(self, dt, workflow, plugin_class_name, plugins_conf, *args, **kwarg
         self.workflow = workflow
         super(BuildPluginsRunner, self).__init__(plugin_class_name, plugins_conf, *args, **kwargs)
 
+    def on_plugin_failed(self):
+        self.workflow.build_has_failed = True
```

Comment:
From the point of view of an exit plugin wanting to know whether everything has gone to plan so far, though, it would be nice for it to have a single thing to check.